addons/registry/2.7.1/install.sh

```diff
@@ -14,6 +14,19 @@ function registry() {
         insert_patches_strategic_merge "$DIR/kustomize/registry/kustomization.yaml" patch-deployment-velero.yaml
         render_yaml_file "$DIR/addons/registry/2.7.1/tmpl-configmap-velero.yaml" > "$DIR/kustomize/registry/configmap-velero.yaml"
         insert_resources "$DIR/kustomize/registry/kustomization.yaml" configmap-velero.yaml
+
+        # Start migration only when usePvcStorage flag is enabled in the installer spec
+        if [ "$REGISTRY_USE_PVC_STORAGE" = "1" ]; then
+            cp "$DIR/addons/registry/2.7.1/patch-deployment-migrate-s3.yaml" "$DIR/kustomize/registry/patch-deployment-migrate-s3.yaml"
+            insert_patches_strategic_merge "$DIR/kustomize/registry/kustomization.yaml" patch-deployment-migrate-s3.yaml
+            cp "$DIR/addons/registry/2.7.1/patch-pvc-for-migrate-s3.yaml" "$DIR/kustomize/registry/patch-pvc-for-migrate-s3.yaml"
+            insert_patches_strategic_merge "$DIR/kustomize/registry/kustomization.yaml" patch-pvc-for-migrate-s3.yaml
+            determine_registry_pvc_size
+            render_yaml_file "$DIR/addons/registry/2.7.1/tmpl-persistentvolumeclaim.yaml" > "$DIR/kustomize/registry/persistentvolumeclaim.yaml"
+            insert_resources "$DIR/kustomize/registry/kustomization.yaml" persistentvolumeclaim.yaml
+            render_yaml_file "$DIR/addons/registry/2.7.1/tmpl-configmap-migrate-s3.yaml" > "$DIR/kustomize/registry/configmap-migrate-s3.yaml"
+            insert_resources "$DIR/kustomize/registry/kustomization.yaml" configmap-migrate-s3.yaml
+        fi
     else
```

Comment:
this will migrate the data but it will not reconfigure the registry to use the pvc. i think you may want something like this.

```suggestion
    fi

    if ! registry_pvc_exists && object_store_exists; then
```

Reply:
Also you may want to delete the registry service or something like that so that nothing can write to the registry during the migration

https://github.com/replicatedhq/kURL/blob/master/addons/registry/2.7.1/service.yaml